packages/python/plotly/plotly/express/_core.py

```diff
@@ -265,17 +265,24 @@ def make_trace_kwargs(args, trace_spec, trace_data, mapping_labels, sizeref):
                     attr_value in ["ols", "lowess"]
                     and args["x"]
                     and args["y"]
-                    and len(trace_data) > 1
+                    and len(trace_data[[args["x"], args["y"]]].dropna()) > 1
                 ):
                     import statsmodels.api as sm
 
                     # sorting is bad but trace_specs with "trendline" have no other attrs
                     sorted_trace_data = trace_data.sort_values(by=args["x"])
-                    y = sorted_trace_data[args["y"]]
-                    x = sorted_trace_data[args["x"]]
+                    y = sorted_trace_data[args["y"]].values
+                    x = sorted_trace_data[args["x"]].values
 
+                    x_is_date = False
                     if x.dtype.type == np.datetime64:
                         x = x.astype(int) / 10 ** 9  # convert to unix epoch seconds
+                        x_is_date = True
+                    elif x.dtype.type == np.object_:
+                        x = x.astype(np.float64)
```

Comment:
Here if we have strings like the `data` column of `px.data.stocks()` this will raise an error message `ValueError: could not convert string to float: '2018-01-01'`. This is not a regression since the error message is better than the current one deep in statsmodels, but this line could be in a try except statement with an error message written by us.

Reply:
agreed, will fix